src/realm/dictionary.cpp

```diff
@@ -515,7 +496,10 @@ const Mixed Dictionary::operator[](Mixed key)
         ret = get(key);
     }
     catch (const KeyNotFound&) {
-        create();
+        // FIXME: This may throw `KeyNotFound` again, but for a different
```

Comment:
Should this be fixed? If yes, let's fix it in this PR or file a ticket to follow up on it. Same with the other FIXME-s - we should either file follow-up tickets and add links in the code or have them as regular comments.